### PR TITLE
Configure typings for external use

### PR DESCRIPTION
**Pull request by edgarmueller**
Export typings and put generated source into root folder (also update .gitignore).  This enables API consumers to do imports a-la `import { ... } from "jsonforms"`.

I'm not sure whether all required types/values are exported but I'd add those opportunistically.

### CODE REVIEW

**Comment by edgarmueller** [comment]
@eneufeld I tested by compiling locally and consuming the result in another NPM project.

File: src/components/form/form-directive.ts [hunk 1]
```diff
@@ -166,11 +166,19 @@ export interface JsonFormsDirectiveScope extends ng.IScope {
...
+<div>
+    <form role='form' class='jsf-form rounded'></form>
+</div>`;
 
-export class JsonFormsDirective implements ng.IDirective {
 
+export class JsonFormsDirective implements ng.IDirective {
+    static $inject = ['$templateCache'];
+    constructor($templateCache: ng.ITemplateCacheService) {
+        $templateCache.put('form.html', formTemplate);
```
Comment (eneufeld) [inline src/components/form/form-directive.ts]: why not do this in a run block, just like in all other cases?

**Comment by eneufeld** [comment]
The jsonforms.ts is now a mixup of two schema definitions (ui-schema and json-schema) and our exports, is this a good way to do it?

**Comment by eneufeld** [comment]
Looks good, 
did't you have typescript warnings for using " in the imports instead of ' ?

**Comment by edgarmueller** [comment]
I probably do..will update :sweat_smile:

**Comment by eneufeld** [comment]
looks good, thank you

**Pull request merged**